packages/proxy/lib/http/index.ts

```diff
@@ -429,10 +430,13 @@ export class Http {
     }
   }
 
-  reset () {
+  reset (options: { resetPrerequests?: boolean } = {}) {
     this.buffers.reset()
     this.setAUTUrl(undefined)
-    this.preRequests.reset()
+
+    if (options.resetPrerequests) {
```

Comment:
can you add a not on when one would not want to reset prerequests?

Reply:
actually, is adding this option needed? it seems this reset is only called in one place within the server code

Reply:
NetworkProxy.reset is called:

* in between tests - https://github.com/cypress-io/cypress/blob/develop/packages/server/lib/server-base.ts#L475
* in between specs - https://github.com/cypress-io/cypress/blob/develop/packages/server/lib/server-base.ts#L620
* onResolveUrl - https://github.com/cypress-io/cypress/blob/develop/packages/server/lib/server-base.ts#L732

We only want to reset the prerequests in between specs.

Reply:
Resolved here: [`04617eb` (#28517)](https://github.com/cypress-io/cypress/pull/28517/commits/04617eb1b3565c380ac89d12f575f2e731cba4f6)

Reply:
Ahh I searched on just networkProxy.reset() and didn't check for networkProxy **?** .reset()

Reply:
Why do we only care about resetting prerequests once per spec now? Do we really only care about this for resetting the service worker manager?

Reply:
This is something that was added somewhat recently (and incorrectly). When test isolation is off for example, we wouldn't want to just drop requests that were happening in a previous test. Only when we start a fresh new session in a new browser tab do we want to reset them.

Reply:
Gottchya - Sound this be called out in the changelog as well? do we have an issue for this?

Reply:
I can add a comment in the changelog. There's no issue for this, it's just something I found in the process of fixing this. I can add an issue though.